cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -305,7 +306,14 @@ private BodyConsumer deployApplication(final HttpResponder responder,
                                          final String archiveName,
                                          final String configString) throws IOException {
 
-    Location namespaceHomeLocation = namespacedLocationFactory.get(namespace);
+    Location namespaceHomeLocation = null;
+    try {
+      namespaceHomeLocation = namespacedLocationFactory.get(namespace);
+    } catch (NamespaceNotFoundException e) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, e.getMessage());
+    } catch (UnauthenticatedException e) {
+      responder.sendString(HttpResponseStatus.FORBIDDEN, e.getMessage());
```

Comment:
Why need to catch explicitly? Isn't that we have an exception handler that will inspect the exception and responds appropriately?